src/NuGet.Core/NuGet.Commands/PublicAPI/net472/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1 @@
+static NuGet.Commands.MSBuildProjectFrameworkUtility.GetProjectFramework(string projectFilePath, string targetFrameworkMoniker, string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion, string targetPlatformMinVersion) -> NuGet.Frameworks.NuGetFramework
```

Comment:
Maybe this is being too idealistic, and I know that this class is already public in this assembly. But, it doesn't feel like it fits here. `NuGet.Commands` mostly is about classes that represent a user action (`AddSourceArgs`, `RestoreArgs`) and their runners (`AddSourceCommand`, `RestoreCommand`). This utility method doesn't feel right.

Looking at assembly names, `NuGet.ProjectModel` seems more appropriate, although looking at the package description in the csproj, not so much. `NuGet.Common`'s naming sounds like a dumping ground, but given this utility is used by MSBuild restore, static graph restore, (nuget.exe?), VS restore, a generic common assembly also sounds reasonable.

Reply:
You used the word restore in every one of your examples, and much of that lives in NuGet.Commands :)

I think moving things down the stack requires a very convincing argument. 
Lower down the dependencies == More dependants == More potential to break. 

I know my argument is not perfect, but I don't want to overthink it. We have this class and many utility methods already and I'm not sure it's a deal breaker. 

Happy to discuss further.

Reply:
It's not a deal-breaker. In fact, I don't expect that we will move this new method out of this class/assembly.  But I think we should be more deliberate about API design in general.

All the usages of this helper method are indeed related to restore, and `RestoreCommand` and `RestoreArgs` does live rightfully in `NuGet.Commands`. But `NuGet.Commands` is a higher level of abstraction. It's about executing commands. If I'm writing my little app and I want to be able to figure out the correct `NuGetFramework` for a given project, `NuGet.Commands` doesn't seem like the correct assembly to reference in my app to get this functionality, as my app isn't executing NuGet commands, I'm just trying to figure out the framework.

Reply:
> But I think we should be more deliberate about API design in general.

I agree. 
Honestly, I hate the fact that we have this utility.
Definitely one of the reasons that gives me a pause right now as far as moving it down the stack